components/builder-jobsrv/src/config.rs

```diff
@@ -64,12 +71,29 @@ impl ConfigFile for Config {
         try!(toml.parse_into("cfg.worker_command_addr", &mut cfg.worker_command_addr));
         try!(toml.parse_into("cfg.worker_heartbeat_addr", &mut cfg.worker_heartbeat_addr));
         try!(toml.parse_into("cfg.datastore_addr", &mut cfg.datastore_addr));
-        try!(toml.parse_into("cfg.shards", &mut cfg.shards));
+        try!(toml.parse_into("cfg.datastore_retry_ms", &mut cfg.datastore_retry_ms));
+        try!(toml.parse_into("cfg.pool_size", &mut cfg.pool_size));
         try!(toml.parse_into("cfg.heartbeat_port", &mut cfg.heartbeat_port));
+        try!(toml.parse_into("cfg.shards", &mut cfg.shards));
+        try!(toml.parse_into("cfg.worker_threads", &mut cfg.worker_threads));
         Ok(cfg)
     }
 }
 
+impl DataStoreCfg for Config {
```

Comment:
I really like this pattern. It makes it easy to say that our 'Config' is valuable in many contexts, while making the contract explicit rather than implicit on methods.